tests/submitTransactionPage.test.ts

```diff
@@ -0,0 +1,611 @@
+
+      const methods = await submitMethodsDropdown
+        .locator(".SubmitTx__floater__item__title")
+        .allTextContents();
+
+      expect(methods).toEqual(["via RPC", "via Horizon"]);
+
+      await submitMethodsDropdown.getByText("via Horizon").click();
+
+      const submitTxBtn = page.getByRole("button", {
+        name: "Submit transaction",
+      });
+
+      // Mock the horizon submission api call
+      await page.route("*/**/transactions", async (route) => {
+        await route.fulfill({
+          status: 400,
+          contentType: "application/json",
+          body: JSON.stringify(MOCK_FAILED_TX_HORIZON_RESPONSE),
+        });
+      });
+
+      const responsePromise = page.waitForResponse(
+        (response) =>
+          response.url().includes("/transactions") && response.status() === 400,
+      );
+
+      await submitTxBtn.click();
+
+      await responsePromise;
+
+      await expect(validationCard).toBeVisible();
+
+      await expect(
+        validationCard.getByText("Transaction failed!"),
+      ).toBeVisible();
+
+      await expect(
+        validationCard.getByText("Request failed with status code 400"),
+      ).toBeVisible();
+
+      const txResponseError = validationCard.getByTestId(
+        "submit-tx-horizon-error-extras",
+      );
+      expect(txResponseError.locator(".TxResponse__value")).toBeVisible;
+    });
+
+    test("Submit via RPC", async ({ page }) => {
+      const xdrInput = page.getByLabel(
+        "Input a base-64 encoded TransactionEnvelope",
+      );
+      const validationCard = page.locator(".ValidationResponseCard");
+      const errorValidationCard = page.getByTestId("submit-tx-rpc-error");
+
+      await xdrInput.fill(MOCK_VALID_FAILED_TX_XDR.XDR);
+
+      // Submit, Simulate, and Save TX buttons to be disabled
+      for (const btn of await page
+        .getByRole("button", { name: /transaction/i })
+        .all())
+        await btn.isEnabled();
+
+      expect(page.getByLabel("Transaction hash")).toBeVisible;
+
+      await testSuccessHashAndJson({
+        page,
+        hash: MOCK_VALID_FAILED_TX_XDR.hash,
+        json: MOCK_VALID_FAILED_TX_XDR.JSON,
+      });
+
+      const submitMethodsBtn = page
+        .locator(".SubmitTx__buttons")
+        .getByRole("button", { name: /via/i });
+
+      const submitMethodsDropdown = page.getByTestId(
+        "submit-tx-methods-dropdown",
+      );
+
+      await submitMethodsBtn.click();
+      await expect(submitMethodsDropdown).toBeVisible();
+
+      const methods = await submitMethodsDropdown
+        .locator(".SubmitTx__floater__item__title")
+        .allTextContents();
+
+      expect(methods).toEqual(["via RPC", "via Horizon"]);
+
+      await submitMethodsDropdown.getByText("via RPC").click();
+
+      const submitTxBtn = page.getByRole("button", {
+        name: "Submit transaction",
+      });
+
+      // Mock the rpc submission api call
+      await page.route("**", async (route) => {
```

Comment:
Should we use `https://soroban-testnet.stellar.org/` instead here?

Reply:
yes! it was returning an error with it when I had the response in place but now that I've removed it, I will put it back